### DISCUSSION TITLE
What is the domain of my vercel function

### DISCUSSION THREAD

**Discussion by 0xbe1** [vercel/next.js#47496] [Help]
### Summary

I have a backend service A that my vercel function B is calling into. I want to whitelist the domain of B in A so that A cannot be accessed unexpectedly.

However, I am not able to figure out what's the domain for my vercel function B. let's say my nextjs project is deployed at my-project.vercel.app, whitelisting the domain my-project.vercel.app, A still doesn't actually allow requests from B.

So I believe my vercel function B has a different domain than my-project.vercel.app. What's that? Any help? Thank you!

**Comment by icyJoseph** [comment]
Isn't it like the Vercel functions run serverless tasks, so in a sense, it is the same as a node.js script running locally. So no domain would be implied. In other words, to make a request to a server, you don't need to setup a web server, you just fly the request, to serve incoming requests, yes you need to have server. 

Maybe it is best to use some kind of private/public key between the services.

**Comment by razgraf** [comment]
I've experienced a similar problem recently trying to make a call to a [subgraph](https://thegraph.com/) from a set of `/api` functions (A calls B, B calls GraphQL endpoint). Needless to say, the final endpoint was protected by a list of whitelisted domains, which included the one where A and B were hosted, but it still returned an unauthorized response.

My solution was to pass an `{Origin: domain.com}` request header to my GraphQL query in route B. Works with Postman too. I'm not 100% sure why it does (I thought forging the Origin was supposed to be harder) but this is how I got it to work.
